apps/server/src/modules/h5p-editor/libraryStorage/library.entity.ts

```diff
@@ -0,0 +1,288 @@
+		this.machineName = machineName;
+		this.majorVersion = majorVersion;
+		this.minorVersion = minorVersion;
+	}
+}
+
+export class FileMetadata implements IFileStats {
+	name: string;
+
+	birthtime: Date;
+
+	size: number;
+
+	constructor(name: string, birthtime: Date, size: number) {
+		this.name = name;
+		this.birthtime = birthtime;
+		this.size = size;
+	}
+}
+
+@Entity({ tableName: 'h5p_library' })
+export class InstalledLibrary extends BaseEntity implements IInstalledLibrary {
+	@Property()
+	machineName: string;
+
+	@Property()
+	majorVersion: number;
+
+	@Property()
+	minorVersion: number;
+
+	@Property()
+	patchVersion: number;
+
+	/**
+	 * Addons can be added to other content types by
+	 */
+	@Property()
+	addTo?: {
+		content?: {
+			types?: {
+				text?: {
+					/**
+					 * If any string property in the parameters matches the regex,
+					 * the addon will be activated for the content.
+					 */
+					regex?: string;
+				};
+			}[];
+		};
+		/**
+		 * Contains cases in which the library should be added to the editor.
+		 *
+		 * This is an extension to the H5P library metadata structure made by
+		 * h5p-nodejs-library. That way addons can specify to which editors
+		 * they should be added in general. The PHP implementation hard-codes
+		 * this list into the server, which we want to avoid here.
+		 */
+		editor?: {
+			/**
+			 * A list of machine names in which the addon should be added.
+			 */
+			machineNames: string[];
+		};
+		/**
+		 * Contains cases in which the library should be added to the player.
+		 *
+		 * This is an extension to the H5P library metadata structure made by
+		 * h5p-nodejs-library. That way addons can specify to which editors
+		 * they should be added in general. The PHP implementation hard-codes
+		 * this list into the server, which we want to avoid here.
+		 */
+		player?: {
+			/**
+			 * A list of machine names in which the addon should be added.
+			 */
+			machineNames: string[];
+		};
+	};
+
+	/**
+	 * If set to true, the library can only be used be users who have this special
+	 * privilege.
+	 */
+	@Property()
+	restricted: boolean;
+
+	@Property()
+	author?: string;
+
+	/**
+	 * The core API required to run the library.
+	 */
+	@Property()
+	coreApi?: {
+		majorVersion: number;
+		minorVersion: number;
+	};
+
+	@Property()
+	description?: string;
+
+	@Property()
+	dropLibraryCss?: {
+		machineName: string;
+	}[];
+
+	@Property()
+	dynamicDependencies?: LibraryName[];
+
+	@Property()
+	editorDependencies?: LibraryName[];
+
+	@Property()
+	embedTypes?: ('iframe' | 'div')[];
+
+	@Property()
+	fullscreen?: 0 | 1;
+
+	@Property()
+	h?: number;
+
+	@Property()
+	license?: string;
+
+	@Property()
+	metadataSettings?: {
+		disable: 0 | 1;
+		disableExtraTitleField: 0 | 1;
+	};
+
+	@Property()
+	preloadedCss?: Path[];
+
+	@Property()
+	preloadedDependencies?: LibraryName[];
+
+	@Property()
+	preloadedJs?: Path[];
+
+	@Property()
+	runnable: boolean | 0 | 1;
```

Comment:
is it a good choice to making it falsy, truthy as well as allow 0 or 1? can that be solved in a better way?